parquet-scrooge/src/main/java/parquet/scrooge/ScroogeStructConverter.java

```diff
@@ -273,11 +261,45 @@ private ThriftType convertClassToThriftType(Class typeClass) {
       return new ThriftType.I64Type();
     } else if (typeClass == String.class) {
       return new ThriftType.StringType();
+    } else if (typeClass == scala.collection.Seq.class){
+      Manifest<?> a = typeManifest.typeArguments().apply(0);
+      return convertListTypeField(name, a, requirement);
+    } else if (typeClass == scala.collection.Set.class){
+      Manifest<?> setElementManifest = typeManifest.typeArguments().apply(0);
+      return convertSetTypeField(name, setElementManifest, requirement);
+    } else if (typeClass == scala.collection.Map.class){
+      List<Manifest<?>> ms = JavaConversions.seqAsJavaList(typeManifest.typeArguments());
+      Manifest keyManifest = ms.get(0);
+      Manifest valueManifest = ms.get(1);
+      return convertMapTypeField(name, keyManifest, valueManifest, requirement);
+    } else if (com.twitter.scrooge.ThriftEnum.class.isAssignableFrom(typeClass)){
+      return convertEnumTypeField(typeClass, name);
     } else {
       return convertStructFromClass(typeClass);
     }
   }
 
+  /**
+   * There is a bug/inconsistency in ThriftStructConverter:
+   * The field name of a nested list/set has the same name as its parent. for example:
+   *  1: required list<list<int>> lm
+   * the name of the inner list is lm, but is should be lm_list_elem.
+   *
+   * This bug/inconsistency does not affect the converted parquet schema because ThriftSchemaConverter ignores the name
+   * of inner list.
+   *
+   * Here in ScroogeStructConverter the same behavior is replicated to generate the same ThriftStruct with
+   * ThriftStructConverter
+   *
+   * TODO: This fix might not be necessary since parquet mostly operates on MessageType, the ThriftStruct is an intermediate
+   * layer for generating MessageType from a thrift/scrooge class. The fix could be removed once we are confident it's
+   * safe for doing so.
+   * @return
+   */
+  private String fixNestListOrSetName(String name) {
+    return name.replaceAll("_list_elem|_set_elem", "");
```

Comment:
If we can fix the bug I agree, lets do that.

But if we want this behavior, what I mean is, instead of doing a replace, can't we just use `fieldName` instead of `fieldName + "_list_elem"` above? I assume that's what the TBase impl does?